src/Neo/SmartContract/Manifest/ExtendedType.cs

```diff
@@ -0,0 +1,331 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// ExtendedType.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Json;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+
+namespace Neo.SmartContract.Manifest
+{
+#nullable enable
+
+    public class ExtendedType : IInteroperable, IEquatable<ExtendedType>
+    {
+        /// <summary>
+        /// The type of the parameter. It can be any value of <see cref="ContractParameterType"/> except <see cref="ContractParameterType.Void"/>.
+        /// </summary>
+        public ContractParameterType Type { get; set; }
+
+        /// <summary>
+        /// NamedType is used to refer to one of the types defined in the namedtypes object of Contract,
+        /// so namedtypes MUST contain a field named name.
+        /// This field is only used for structures (ordered set of named values of different types),
+        /// if used other fields MUST NOT be set, except for the type which MUST be an Array.
+        /// Value string MUST start with a letter and can contain alphanumeric characters and dots.
+        /// It MUST NOT be longer than 64 characters.
+        /// </summary>
+        public string? NamedType { get; set; }
```

Comment:
One thing that is missing from the current PR is an extension of contract ABI: it should include additional `namedtypes` field that contains the definition of all extended types used by this contract. Ref. https://github.com/neo-project/proposals/blob/master/nep-25.mediawiki#contract.